src/client.js

```diff
@@ -9,26 +9,61 @@ export default class Client {
     this.engineName = engineName
     this.apiEndpoint = `https://${accountHostKey}.api.swiftype.com/api/as/v1/`
     this.searchPath = `engines/${this.engineName}/search`
+    this.clickPath = `engines/${this.engineName}/click`
   }
 
   /**
-   * Send a search request to the Swiftype App Search Api
-   * https://swiftype.com/documentation/app-search/
+   * Sends a search request to the Swiftype App Search Api
    *
    * @param {String} query String, Query, or Object that is used to perform a search request.
    * @param {Object} options Object used for configuring the search like search_fields and result_fields
    * @returns {Promise<ResultList>} a Promise that returns a {ResultList} when resolved, otherwise throws an Error.
    */
   search(query, options) {
     const params = Object.assign({ query: query }, options)
-    return this._request(`${this.searchPath}.json`, params).then(({ response, json }) => {
+    return this._requestJSON(`${this.searchPath}.json`, params).then(({ response, json }) => {
       if (!response.ok) {
-        throw new Error(`[${response.status}] ${json.errors}`)
+        throw new Error(`[${response.status}]${json.errors ? ' ' + json.errors : ''}`)
       }
       return new ResultList(json.results, _omit(json, 'results'))
     })
   }
 
+  /**
+   * Sends a click event to the Swiftype App Search Api, to track a click-through event
+   *
+   * @param {String} query Query that was used to perform the search request
+   * @param {String} documentId ID of the document that was clicked
+   * @param {String} requestId Request_id from search response
+   * @param {String[]} tags Unique tags to identify this request in the Dashboard
```

Comment:
these tags aren't meant to be unique - they're meant to be more for categorizing requests (for example tracking clicks on mobile devices vs desktop devices, things like that)

Reply:
Ah, ok, thanks. I'll update that.